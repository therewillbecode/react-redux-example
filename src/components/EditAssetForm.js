import React from "react";
import { Button, Form, Segment } from "semantic-ui-react";

const EditAssetForm = ({
  dispatched,
  dispatchAsset,
  handleSubmit,
  handleNameChange,
  handleCommentChange,
  name,
  id,
  comment
}) => (
  <div
    style={{
      width: "30%",
      left: "35%",
      position: "relative",
      padding: "3em"
    }}
  >
    <Segment inverted color="grey" textAlign="center">
      <Form inverted onSubmit={handleSubmit}>
        <h2>Edit Asset ID: {id} </h2>
        <Form.Group>
          <Form.Field width="4">
            <div style={{ marginTop: "1.6em" }}>
              <label>Edit Name</label>
              <Form.Input
                disabled={dispatched}
                name="name"
                value={name}
                onChange={handleNameChange}
              />
            </div>
          </Form.Field>
          <Form.Field width="8">
            <label>Edit Comment</label>
            <Form.TextArea
              name="comment"
              value={comment}
              onChange={handleCommentChange}
            />
          </Form.Field>
          <Form.Field width="4">
            <div style={{ marginTop: "2.5em" }}>
              <Form.Button inverted primary content="Update Asset" />
            </div>
          </Form.Field>
        </Form.Group>
        <Button
          onClick={dispatchAsset}
          icon={dispatched ? "ship" : null}
          disabled={dispatched}
          primary
          content={dispatched ? "Dispatched" : "Dispatch Asset"}
        />
      </Form>
    </Segment>
  </div>
);

export default EditAssetForm;
