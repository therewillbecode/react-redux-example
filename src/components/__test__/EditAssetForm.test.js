/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";

import EditAssetForm from "../EditAssetForm";

describe("EditAssetForm", () => {
  beforeEach(() => {});
  it("renders correctly", () => {
    const tree = renderer.create(<EditAssetForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
