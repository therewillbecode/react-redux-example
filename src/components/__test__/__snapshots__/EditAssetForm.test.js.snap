// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAssetForm renders correctly 1`] = `
<div
  style={
    Object {
      "left": "35%",
      "padding": "3em",
      "position": "relative",
      "width": "30%",
    }
  }
>
  <Segment
    color="grey"
    inverted={true}
  >
    <Form
      as="form"
      inverted={true}
      onSubmit={undefined}
    >
      <h2
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Edit Asset ID: 
         
      </h2>
      <FormGroup>
        <FormField
          width="4"
        >
          <div
            style={
              Object {
                "marginTop": "1.6em",
              }
            }
          >
            <label>
              Edit Name
            </label>
            <FormInput
              as={[Function]}
              control={[Function]}
              disabled={undefined}
              name="name"
              onChange={undefined}
              value={undefined}
            />
          </div>
        </FormField>
        <FormField
          width="8"
        >
          <label>
            Edit Comment
          </label>
          <FormTextArea
            as={[Function]}
            control={[Function]}
            name="comment"
            onChange={undefined}
            value={undefined}
          />
        </FormField>
        <FormField
          width="4"
        >
          <div
            style={
              Object {
                "marginTop": "2em",
              }
            }
          >
            <FormButton
              as={[Function]}
              content="Update Asset"
              control={[Function]}
              inverted={true}
              primary={true}
              size="large"
            />
          </div>
        </FormField>
      </FormGroup>
      <Segment
        color="grey"
        inverted={true}
      >
        <div
          style={
            Object {
              "left": "32%",
              "position": "relative",
            }
          }
        >
          <Button
            as="button"
            content="Dispatch Asset"
            disabled={undefined}
            icon={null}
            onClick={undefined}
            primary={true}
            size="huge"
          />
        </div>
      </Segment>
    </Form>
  </Segment>
</div>
`;
